### ISSUE TITLE
[Regression] JSON conflicts with AWS SDK

### ISSUE THREAD

**Issue by jonlambert** [fastlane/fastlane#7707]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

When running fastlane, using the S3 uploader action, I get the following error:

```
/usr/local/Cellar/ruby/2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:2284:in `raise_if_conflicts': Unable to activate aws-sdk-v1-1.66.0, because json-2.0.2 conflicts with json (~> 1.4) (Gem::ConflictError)
from /usr/local/Cellar/ruby/2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:1407:in `activate'
from /usr/local/Cellar/ruby/2.3.1/lib/ruby/2.3.0/rubygems.rb:196:in `rescue in try_activate'
from /usr/local/Cellar/ruby/2.3.1/lib/ruby/2.3.0/rubygems.rb:193:in `try_activate'
from /usr/local/Cellar/ruby/2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:125:in `rescue in require'
from /usr/local/Cellar/ruby/2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:40:in `require'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/actions/s3.rb:221:in `load_from_v1_gem_name'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/actions/s3.rb:247:in `s3_client'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/actions/s3.rb:70:in `run'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/runner.rb:232:in `block (2 levels) in execute_action'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/runner.rb:217:in `block in execute_action'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/runner.rb:213:in `chdir'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/runner.rb:213:in `execute_action'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/runner.rb:135:in `trigger_action_by_name'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/fast_file.rb:146:in `method_missing'
from Fastfile:50:in `block (2 levels) in parsing_binding'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/lane.rb:33:in `call'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/lane_manager.rb:52:in `cruise_lane'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/command_line_handler.rb:30:in `handle'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/commands_generator.rb:96:in `block (2 levels) in run'
from /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.3/lib/commander/command.rb:178:in `call'
from /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.3/lib/commander/command.rb:153:in `run'
from /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.3/lib/commander/runner.rb:446:in `run_active_command'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:38:in `run!'
from /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.3/lib/commander/delegates.rb:15:in `run!'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/commands_generator.rb:293:in `run'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/commands_generator.rb:36:in `start'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/fastlane/lib/fastlane/cli_tools_distributor.rb:59:in `take_off'
from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.3.1/bin/fastlane:5:in `<top (required)>'
from /usr/local/bin/fastlane:23:in `load'
from /usr/local/bin/fastlane:23:in `<main>'
```

### Captured Output

Command Used: `upload`
<details><summary>Output/Log</summary>

```

[18:36:22]: It seems like you wanted to load some plugins, however they couldn't be loaded
[18:36:22]: Please follow the troubleshooting guide: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/PluginsTroubleshooting.md
[18:36:22]: -------------------------------------------------
[18:36:22]: --- Step: Verifying required fastlane version ---
[18:36:22]: -------------------------------------------------
[18:36:22]: Your fastlane version 2.3.1 matches the minimum requirement of 1.111.0  ✅
[18:36:22]: ------------------------------
[18:36:22]: --- Step: default_platform ---
[18:36:22]: ------------------------------
[18:36:22]: Driving the lane 'ios upload' 🚀
#########[18:36:22]: ----------------
[18:36:22]: --- Step: s3 ---
[18:36:22]: ----------------
+------------------+------------+
|         Lane Context          |
+------------------+------------+
| DEFAULT_PLATFORM | ios        |
| PLATFORM_NAME    | ios        |
| LANE_NAME        | ios upload |
+------------------+------------+

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 3    | s3                                  | 0           |
+------+-------------------------------------+-------------+

[18:36:22]: fastlane finished with errors


```

</details>

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                 | Value                                       |
| ------------------- | ------------------------------------------- |
| OS                  | 10.12                                       |
| Ruby                | 2.3.1                                       |
| Bundler?            | false                                       |
| Git                 | git version 2.9.0                           |
| Installation Source | /usr/local/bin/fastlane                     |
| Host                | Mac OS X 10.12 (16A323)                     |
| Ruby Lib Dir        | /usr/local/Cellar/ruby/2.3.1/lib            |
| OpenSSL Version     | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained        | false                                       |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version       | 8.2.1                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_GB.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "1.111.0"

default_platform :ios

platform :ios do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
  end

  desc "Runs all the tests"
  lane :test do
    scan
  end

  desc "Submit a new Beta Build to Apple TestFlight"
  desc "This will also make sure the profile is up to date"
  lane :beta do
    # match(type: "appstore") # more information: https://codesigning.guide
    gym # Build your app - more options available
    pilot

    # sh "your_script.sh"
    # You can also use other beta testing services here (run `fastlane actions`)
  end

  desc "Build and deploy an ad-hoc version of the app"
  lane :release do
    increment_build_number
    cert
    sigh(adhoc: true)
    gym
  end

  desc "Upload the latest ad-hoc build to the S3 bucket"
  lane :upload do
    print ENV['S3_ACCESS_KEY']
    s3(
      access_key: ENV['S3_ACCESS_KEY'],               # Required from user.
      secret_access_key: ENV['S3_SECRET_ACCESS_KEY'], # Required from user.
      bucket: <redacted>,                            # Required from user.
      ipa: <redacted>,
      dsym: <redacted>,
      path: <redacted>,
      upload_metadata: true,                          # Upload version.json, plist and HTML. Set to false to skip uploading of these files.
    )
  end

  desc "Grabs a new cert & provisioning profile" 
  lane :dev do
    # 1
    cert(development: true)
    # 2
    sigh(development: true)
  end

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.3.1   | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.4        |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.2        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.2        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.4        |
| addressable               | 2.5.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| babosa                    | 1.0.2        |
| highline                  | 1.7.8        |
| commander                 | 4.4.3        |
| excon                     | 0.54.0       |
| faraday                   | 0.10.0       |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161129 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.0.1        |
| gh_inspector              | 1.0.2        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.9.20       |
| json                      | 2.0.2        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.1.1        |
| bundler                   | 1.13.7       |
| faraday_middleware        | 0.10.1       |
| io-console                | 0.4.5        |
</details>

*generated on:* **2016-12-30**
</details>

**Comment by epankratov** [comment]
We have the same problem: in order to use S3 upload action, our code uses aws-sdk-v1-1.66.0 instead of newer one. 3 days ago, it began to raise the conflict with json gem installed in the system.

**Comment by pipacs** [comment]
Same here, with fastlane-2.3.1, which depends on json-2.x which is apparently not liked by aws-sdk-v1...

**Comment by KrauseFx** [comment]
Hey everyone, we just pushed [fastlane 2.4.0](https://github.com/fastlane/fastlane/releases/tag/2.4.0) which fixes the bundled _fastlane_, just run `fastlane update_fastlane` to get the latest release 🚀

However this release breaks support for Ruby 2.4.0 for now, which we should be able to add again soon. For now, please downgrade to an older version of Ruby, or use the bundled _fastlane_ from [download.fastlane.tools](https://download.fastlane.tools)

Please let us know if that works for you 👍

**Comment by waltermvp** [comment]
@KrauseFx to the rescue 2.4.0 fixed my issue.

If you:
* use ruby 2.3.x
* use aws-sdk-v1-1.66.0 


if you have `nokogiri` issues this post helped http://www.nokogiri.org/tutorials/installing_nokogiri.html#mac_os_x (which I found here https://github.com/sparklemotion/nokogiri/issues/1483)

**Comment by epankratov** [comment]
@KrauseFx Yes, it works fine now. Thank you

**Comment by KrauseFx** [comment]
Thanks everyone 👍

**Comment by pipacs** [comment]
Thank you Felix, 2.4.0 worked, however I had to uninstall json-2.0.2 before running the update:

  sudo gem uninstall json -v2.0.2
  sudo fastlane update_fastlane

**Comment by samuelpadua-luizalabs** [comment]
@pipacs This worked for me. Thanks!
